### PR TITLE
APPS 4381 spryk improvements

### PR DESCRIPTION
**Pull request by stereomon**
- Developer(s): @stereomon

- Ticket: https://spryker.atlassian.net/browse/APPS-4381

- Docs PR: ACADEMY_URL_HERE

- Release Group: URL_HERE

- merge: merge

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   SprykSrc              | patch (currently 0.0.x)  |                       |

-----------------------------------------

### Improvements

- Added `APPLICATION_ENV` and `APPLICATION_DEBUG` constants to control the behavior of the Symfony Kernel.
- Improved `preSpryks` and `postSpryks` configuration.
- Added the possibility to use `spryks`  instead of `preSpryks` and `postSpryks` definitions in Wrapper Spryks.
- Allow resources to contain slashes (e.g. /foo/bar) in Glue Spryks.

### Fixes

- Use default `ArrayAdapter` as caching adapter for the Symfony ExpressionLanguage to prevent writing to the cache files within the `spryk.phar` which would cause an exception as it is not possible to write to a PHAR archive.

### Deprecations

- Deprecated `\SprykerSdk\Spryk\Model\Spryk\Filter\NormalizeResourceNameFilter`.

### CODE REVIEW

File: bin/environment.php [hunk 1]
```diff
@@ -49,4 +49,9 @@
 
 define('SPRYK_ROOT_DIR', __DIR__ . '/../');
 
+$applicationEnv = getenv('APPLICATION_ENV');
+$allowedApplicationEnvs = ['dev', 'prod', 'test'];
```
Comment (k4emic) [inline bin/environment.php:53]: Consider moving these to (class?) constants

**Review by k4emic — COMMENTED** [review]
Please check the comments I left. Looks good otherwise.

File: config/services.php [hunk 2]
```diff
@@ -36,7 +39,8 @@
     $services->load('SprykerSdk\\', '../src/')
         ->exclude('../src/{DependencyInjection,Tests,Kernel.php}');
 
-    $services->set(ExpressionLanguage::class);
+    $services->set(ExpressionLanguage::class)
+        ->args([null, []]);
```
[the remark this answers is not in the document]
Comment (stereomon) [reply config/services.php:50]: Where would you put those?

File: bin/environment.php (hunk 1 above)
Comment (stereomon) [reply bin/environment.php:53]: Where would you put those too?

**Review by k4emic — APPROVED** [review]

**Pull request merged**